lib/web/terminal.go

```diff
@@ -299,6 +299,16 @@ func (t *TerminalHandler) streamTerminal(ws *websocket.Conn, tc *client.Teleport
 	// either an error occurs or it completes successfully.
 	err := tc.SSH(t.terminalContext, t.params.InteractiveCommand, false)
 
+	// Handle remote processes that exit with error codes, eg: RemoteCommandFailure (255).
+	if t.sshSession != nil {
+		if err := t.sshSession.Wait(); err != nil {
+			if exitErr, ok := err.(*ssh.ExitError); ok {
+				t.log.Warnf("Remote shell exited with error code: %v", exitErr.ExitStatus())
+				return
+			}
+		}
+	}
+
```

Comment:
Should this happen after the error from `tc.SSH` is handled?
There's a fallback to a different `tc.SSH` call below for example.

Reply:
revised and tested: https://github.com/gravitational/teleport/pull/4088/commits/9fe8b7c148ae9c30e3d81adead0b32f171b03433